apps/root/src/state/balances/actions.ts

```diff
@@ -1,121 +1,130 @@
-import { createAction, createAsyncThunk } from '@reduxjs/toolkit';
-import { CurrentPriceForChainResponse, Token, TokenList } from '@types';
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { ChainId, CurrentPriceForChainResponse, Token, TokenList } from '@types';
 import { BalancesState, TokenBalancesAndPrices } from './reducer';
 import { ExtraArgument, RootState } from '@state';
 import { BigNumber } from 'ethers';
 import { NETWORKS } from '@constants';
 import { flatten, keyBy, set } from 'lodash';
-
-export const setLoadingChainPrices = createAction<{ chainId: number; isLoading: boolean }>(
-  'balances/setLoadingChainPrices'
-);
+import { PROTOCOL_TOKEN_ADDRESS, getProtocolToken } from '@common/mocks/tokens';
 
 export const fetchWalletBalancesForChain = createAsyncThunk<
   { chainId: number; tokenBalances: TokenBalancesAndPrices },
   { tokenList: TokenList; chainId: number; walletAddress: string },
   { extra: ExtraArgument }
->('balances/fetchBalancesForChain', async ({ tokenList, chainId, walletAddress }, { extra: { web3Service } }) => {
+>('balances/fetchWalletBalancesForChain', async ({ tokenList, chainId, walletAddress }, { extra: { web3Service } }) => {
   const sdkService = web3Service.getSdkService();
   const tokens = Object.values(tokenList);
 
-  let balances: Record<number, Record<string, BigNumber>> = {};
-  try {
-    balances = await sdkService.getMultipleBalances(tokens, walletAddress);
-  } catch (e) {
-    console.error(e);
-  }
+  const balances = await sdkService.getMultipleBalances(tokens, walletAddress);
 
-  const tokenAccountBalances: TokenBalancesAndPrices = {};
-  Object.entries(balances[chainId]).forEach(([tokenAddress, balance]) => {
+  const tokenBalances = Object.entries(balances[chainId]).reduce((acc, [tokenAddress, balance]) => {
```

Comment:
you can use `reduce<TokenBalancesAndPrices>` and will have no need for the bottom `as` plus it will typecheck the reducer function return value